packages/salesforcedx-vscode-core/test/vscode-integration/commands/forceAuthDevHub.test.ts

```diff
@@ -61,4 +61,34 @@ describe('Command chosen is based on results of isDemoMode()', () => {
     process.env.SFDX_ENV = '';
     expect(createExecutor() instanceof ForceAuthDevHubExecutor).to.be.true;
   });
+
+  describe('Should work correctly in container mode', () => {
+    afterEach(() => {
+      delete process.env.SFDX_CONTAINER_MODE;
+    });
+    it('and use force:auth:web:login when container mode is not defined', () => {
```

Comment:
I don't think we use `and` to denote the `it(...)`.

I'm open to discuss this but it's better to follow the established convention of a file.

Reply:
If you see these nested statements when running npm test it makes more sense....we're kinda building up broken sentences.  At least that's what i've done with testing in the past like this.
```
Command chosen is based on results of isDemoMode()
    ✓ Should use ForceAuthDevHubDemoModeExecutor if demo mode is true
    ✓ Should use ForceAuthDevHubExecutor if demo mode is false
    Should work correctly in container mode
      ✓ and use force:auth:web:login when container mode is not defined
      ✓ and use force:auth:device:login when container mode is defined
```
I'll admit that I should adjust the top level statement.  So that the test output reads like.
```
The command that is run
    should handle isDemoMode()
    ✓ and use ForceAuthDevHubDemoModeExecutor if demo mode is true
    ✓ and use ForceAuthDevHubExecutor if demo mode is false
    should work correctly in container mode
      ✓ and use force:auth:web:login when container mode is not defined
      ✓ and use force:auth:device:login when container mode is defined
```

But to fit in, I'll try to match the current implementation.

Reply:
@lcampos Thoughts?

Reply:
IMO, I can see how the `it(...)` statements are supposed to fit together to complete a sentence and I like the idea of continuity when running all the tests together. However I just don't often read the output of the mocha runner form top to bottom like a complete paragraph or sentence but more like a list of bullet points and zero in on a single line or test.
With that in mind I would like to be able to understand what failed to work in as few words as possible.

Reply:
The current naming convention follows Spec-style testing, which originated from Behavior Driven Development (BDD). When you write a test you want to specify the behavior of something.

The purpose of `describe` is to group tests that are common by certain criteria and also allows for nesting depending on how specific you want to get with the grouping strategy. In this case I'd say the use of `Should do x ...` in the describe does not provide as much grouping value as something like `In container mode`.

In the spirit of specifying the behavior of something and following BDD, we are using `it` over the `test` interface since we want to describe how the feature should work. If you are reading the test you read it as `it should ...`.

A good example of what I just mentioned are the telemetry tests. The describes are used as ways of grouping tests and the it clauses contain messages that correspond to the grouping they're part of and the expected outcome.

```
  Telemetry 
    in dev mode
      ✓ Should not initialize telemetry reporter
      ✓ Should show telemetry info message
      ✓ Should not show telemetry info message

    production mode
      ✓ Should show telemetry info message
      ✓ Should not show telemetry info message
      ✓ Should send telemetry data
```

This convention only makes sense if you are following BDD practices, which we've standardized on and that's why our mocha runner is configured to only allow bdd interface usage (https://mochajs.org/#interfaces). Let's keep using the current conventions.

Reply:
Cool, thanks for that example @lcampos.  Here's where I ended up.

Force Auth Web Login is based on environment variables
    in demo mode
      ✓ Should use ForceAuthDevHubDemoModeExecutor if demo mode is true
      ✓ Should use ForceAuthDevHubExecutor if demo mode is false
    in container mode
      ✓ Should use force:auth:web:login when container mode is not defined
      ✓ Should use force:auth:web:login when container mode is empty
      ✓ Should use force:auth:device:login when container mode is defined